### ISSUE TITLE
Apply overrides for Ory and Allow type of access when using Istio JWT

### ISSUE THREAD

**Issue by barchw** [kyma-project/api-gateway#75]
**Description**

Since we want to support Ory rules and `allow` access in parallel, we need to add something similar to this (probably as a separate AP resource):
```yaml
---
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: httpbin
  namespace: foo
spec:
  action: ALLOW
  rules:
    - from:
        - source:
            principals:
              - "cluster.local/ns/kyma-system/sa/oathkeeper-maester-account" # For any beside allow
              - "cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account" # For allow handler
      to:
      - operation:
          paths: ["/headers"]
```

**Reasons**

- We need to support parallel exposing of workloads with Ory, as some endpoints use OAuth2 and JWT authorisation in parallel
- This should also provide configuration needed to expose an endpoint with stategy `Allow`

**Sample APIRule that will fail at this moment**

```yaml
apiVersion: gateway.kyma-project.io/v1beta1
kind: APIRule
metadata:
  name: multiple-service
  namespace: $NAMESPACE
  labels:
    app: multiple-service
    example: multiple-service
spec:
  host: multiple-service-example.$DOMAIN_TO_EXPOSE_WORKLOADS
  gateway: $GATEWAY
  service:
    name: httpbin
    port: 8000
  rules:
    - path: /headers
      methods: ["GET"]
      accessStrategies:
        - handler: jwt
    - path: /function
      methods: ["GET"]
      accessStrategies:
        - handler: noop
```

**DoD:**
- [x] provide unit tests

~~- [ ] provide documentation~~ n/a
~~- [ ] test on SKR~~ still on feature branch
~~- [ ] verify [Get started guides](https://kyma-project.io/docs/kyma/latest/02-get-started/)~~ still on feature branch
~~- [ ] verify resource limits~~ still on feature branch
~~- [ ] followup issue~~ still on feature branch
~~- [ ] create release and bump in kyma~~ still on feature branch

**Follow up to**
- https://github.com/kyma-project/api-gateway/pull/53

**Original comment**
- https://github.com/kyma-project/api-gateway/pull/53#discussion_r1009325114

**Related issues**
- https://github.com/kyma-project/api-gateway/issues/34
- https://github.com/kyma-project/api-gateway/issues/96

**Comment by videlov** [comment]
This may be same issue as https://github.com/kyma-project/api-gateway/issues/96
Currently when we have Istio FT on, we don't create Ory Oathkeeper resources, because of:
https://github.com/kyma-project/api-gateway/blob/feature-istio-jwt-handler/controllers/apirule_controller.go#L186-L190
